code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/AbstractAllocator.java

```diff
@@ -228,13 +239,58 @@ protected boolean doAllocate(final AllocationRequest request, final AllocationAt
 
     protected boolean acquireLockAndAllocate(final AllocationRequest request, final AllocationAttempt attempt, Object deallocate) {
         final List<Constraint> finalFailedConstraints = new ArrayList<>();
+        final List<AllocationCandidate> candidatesForCPUMemoryIops = new ArrayList<AllocationCandidate>();
+        final List<AllocationCandidate> candidates = new ArrayList<AllocationCandidate>();
         lockManager.lock(getAllocationLock(request, attempt), new LockCallbackNoReturn() {
             @Override
             public void doWithLockNoResult() {
                 Context c = allocateTimer.time();
                 try {
+                    Iterator<AllocationCandidate> iter = getCandidates(attempt);
+                    while (iter.hasNext()) {
+                        candidatesForCPUMemoryIops.add(iter.next());
+                    }
+                    if (iter != null) {
+                        close(iter);
+                    }
+
+                    Set<Long> alreadyScheduledHostIds = new HashSet<Long>();
+
+                    // scheduler the list and return a list that is able to schedule
+                    for (AllocationCandidate candidate : candidatesForCPUMemoryIops) {
+                        boolean good = true;
+                        
+                        // schedule cpu, memory and iops first
+                        try {
+                            for( Long hostId : candidate.getHosts() ){
+
+                                // in case we already tried to schedule instance for this hostId, then skip it
+                                if (alreadyScheduledHostIds.contains(hostId)) {
+                                    continue;
```

Comment:
I was able to easily reproduce this by creating a host that did not have fio ran on it and then going to the UI and creating a container on that host with the label `io.rancher.resource.read_iops.foo` with a value of `100` to it in the UI.

The container successfully deployed to that host but it should not have been allowed to. When I commented out this if/else block, it failed to schedule, as expected.